packages/nextra-theme-docs/src/components/anchor.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { forwardRef, ComponentProps, ReactElement, Ref } from 'react'
+// eslint-disable-next-line no-restricted-imports -- only in this file we determine either we include <a /> as child of <NextLink /> based of `newNextLinkBehavior` value
+import NextLink, { LinkProps } from 'next/link'
+import { useConfig } from '../config'
+
+type AnchorProps = Omit<ComponentProps<'a'>, 'ref'> &
+  // Use `href` and `onClick` types from <a />
+  Omit<LinkProps, 'href' | 'onClick'> & {
+    newWindow?: boolean
+  }
+
+const Anchor = forwardRef<HTMLAnchorElement, AnchorProps>(function (
+  { href, children, newWindow, ...props },
+  // ref is used in <NavbarMenu />
+  forwardedRef
+): ReactElement | null {
+  const config = useConfig()
+
+  if (!href) {
+    return <a ref={forwardedRef} {...props}>{children}</a>
+  }
+
+  if (newWindow) {
+    return (
+      <a
+        ref={forwardedRef}
+        href={href}
+        target="_blank"
+        rel="noreferrer"
+        aria-selected={false}
+        {...props}
+      >
+        {children}
+      </a>
+    )
+  }
+
+  if (config.newNextLinkBehavior) {
+    return (
+      <NextLink ref={forwardedRef} href={href} {...props}>
+        {children}
+      </NextLink>
+    )
+  }
+
+  return (
+    <NextLink ref={forwardedRef} href={href} passHref>
+      <a {...props}>{children}</a>
+    </NextLink>
+  )
+})
```

Comment:
generic Anchor component that renders either `<NextLink />` with or without `<a />` tag based on next config setting `experimental.newNextLinkBehavior`